keras/engine/topology.py

```diff
@@ -1038,6 +1038,8 @@ def __init__(self, layers=None, mode='sum', concat_axis=-1,
         self.mode = mode
         self.concat_axis = concat_axis
         self.dot_axes = dot_axes
+        if type(self.dot_axes) == int:
+            self.dot_axes = [self.dot_axes, ] * 2
```

Comment:
this was necessary early because `get_output_shape_for` uses this before calling `_validate_arguments`. If you don't like it, we should call `_validate_arguments` early or something like that. The problem with calling `_validate_arguments` I think is that it expects everything to be already built, which is against the lazy initialization required by shape inference, I believe.

But let me know if I'm wrong.